### ISSUE TITLE
Nested functions

### ISSUE THREAD

**Issue by sjanahan** [pg-strom/devel#208]
I am doing some testing with my 9.6 devel instance and I am noticing that the function calls I was hoping to get speed up on are not using the GPU.

---------
I have verified that the PGStrom is installed correctly with the test in the bottom of this inlay 
https://github.com/pg-strom/devel/wiki/Quick-Installation
I get ~3x speed up on those test calls.

--------
There are some unsupported data types in the regular function calls in my DB, such as text and array.

I want to strip out the components of my function calls which use only the supported data types in this wiki list into separate functions to leverage this extension.

My question is PGStrom smart enough to optimize function calls within function calls? Or is it an all or nothing scenario?

For example, say I have a large function that creates variables of various view types, and passes those to an auxiliary function. After that, the auxiliary function calls a series of utility functions that only rely only on the supported data types, such as moving average computations. Would the auxiliary and/or utility functions possibly run on the GPU, or would they not because the entire function was not pg_strom compliant?

**Comment by kaigai** [comment]
> My question is PGStrom smart enough to optimize function calls within function calls? Or is it an all or nothing scenario?

The later is right understanding. PG-Strom tried to generate GPU code if entire expression
node is consist of supported function (or other stuff).

> For example, say I have a large function that creates variables of various view types, and passes
those to an auxiliary function. After that, the auxiliary function calls a series of utility functions that only rely only on the supported data types, such as moving average computations. Would the auxiliary and/or utility functions possibly run on the GPU, or would they not because the entire function was not pg_strom compliant?

Sorry, it looks to me complexity of the above scenario goes beyond our expectation.
Basically, the function call has to be rewritten one formula, and must not contain any
unsupported function and data types.
At least, invocation of utility function are job of CPU. If individual utility function has
part of GPU executable portion, it may make sense....

**Comment by kaigai** [comment]
Anyway, I may be able to comment on if I can see more detailed workloads.

**Comment by kaigai** [comment]
closed due to 3 months from the last activity
